tests/next_tests/integration_tests/feature_tests/iterator_tests/test_horizontal_indirection.py

```diff
@@ -76,7 +76,7 @@ def test_simple_indirection(program_processor):
 
     ref = np.zeros(shape)
     for i in range(shape[0]):
-        ref[i] = inp[i + 1 - 1] if cond[i] < 0 else inp[i + 1 + 1]
+        ref[i] = inp.ndarray[i + 1 - 1] if cond.ndarray[i] < 0 else inp.ndarray[i + 1 + 1]
```

Comment:
```suggestion
        ref[i] = inp[{IDim: i + 1 - 1}] if cond[{IDim: i}] < 0 else inp[{IDim: i + 1 + 1}]
```
I guess we want this.